content/docs/rfcs/36/README.md

```diff
@@ -0,0 +1,809 @@
+- `contentTopic`: The content topic to be set on the message.
+- `version`: The Waku Message version number.
+- `timestamp`: Unix timestamp in nanoseconds.
+
+## Events
+
+Asynchronous events require a callback to be registered.
+An example of an asynchronous event that might be emitted is receiving a message.
+When an event is emitted, this callback will be triggered receiving a JSON string of type `JsonSignal`.
+
+### `JsonSignal` type
+
+```ts
+{
+    nodeId: number;
+    type: string;
+    event: any;
+}
+```
+
+Fields:
+
+- `nodeId`: Integer, go-waku node that emitted the signal.
+- `type`: Type of signal being emitted. Currently, only `message` is available.
+- `event`: Format depends on the type of signal.
+
+For example:
+
+```json
+{
+  "nodeId": 0,
+  "type": "message",
+  "event": {
+    "subscriptionId": 1,
+    "pubsubTopic": "/waku/2/default-waku/proto",
+    "messageId": "0x6496491e40dbe0b6c3a2198c2426b16301688a2daebc4f57ad7706115eac3ad1",
+    "wakuMessage": {
+      "payload": "TODO",
+      "contentTopic": "/my-app/1/notification/proto",
+      "version": 1,
+      "timestamp": 1647826358000000000
+    }
+  }
+}
+```
+
+| `type`    | `event` Type       |
+|:----------|--------------------|
+| `message` | `JsonMessageEvent` | 
+
+### `JsonMessageEvent` type
+
+Type of `event` field for a `message` event:
+
+```ts
+{
+    subscriptionId: number;
+    pubsubTopic: string;
+    messageId: string;
+    wakuMessage: JsonMessage;
+}
+```
+
+- `subscriptionId`: The id of the subscription from which the message was received, returned by [`waku_relay_subscribe`](#extern-char-waku_relay_subscribeint-nodeid-char-pubsubtopic).
+- `pubsubTopic`: The pubsub topic on which the message was received.
+- `messageId`: The message id.
+- `wakuMessage`: The message in [`JsonMessage`](#jsonmessage-type) format.
+
+### `extern void waku_set_event_callback(void* cb)`
+
+Register callback to act as event handler and receive application signals,
+which are used to react to asynchronous events in Waku.
+
+**Parameters**
+
+1. `void* cb`: callback that will be executed when an async event is emitted.
+  The function signature for the callback should be `void myCallback(char* jsonSignal)`
+
+## Node management
+
+### `JsonConfig` type
+
+Type holding a node configuration:
+
+```ts
+interface JsonSignal {
+    host?: string;
+    port?: number;
+    advertiseAddr?: string;
+    nodeKey?: string;
+    keepAliveInterval?: number;
+    relay?: boolean;
+    minPeersToPublish?: number
+}
+```
+
+Fields: 
+
+All fields are optional.
+If a key is `undefined`, or `null`, a default value will be set.
+
+- `host`: Listening IP address.
+  Default `0.0.0.0`.
+- `port`: Libp2p TCP listening port.
+  Default `60000`. 
+  Use `0` for random.
+- `advertiseAddr`: External address to advertise to other nodes.
+  Can be ip4, ip6 or dns4, dns6.
+  Default: ?
```

Comment:
@richard-ramos what is the default value?

Reply:
`null`

Reply:
In the case of `null`, what is the behaviour?